### DISCUSSION TITLE
custom login uri

### DISCUSSION THREAD

**Discussion by wibed** [ory/hydra#2307] [Q&A] [answered]
is it possible to configure a custom login uri for each relaying party?

like redirect_uri, but a string instead of an array

**Comment by vinckr** [comment]
It is a string right?
`"redirect_uris": ["string"]`
https://www.ory.sh/hydra/docs/reference/api/#oauth2client

What are you looking to achieve and why is not working?

**Reply by wibed** [reply]
I try to define a custom login_uri as set by the env var in the dockerfile

`URLS_LOGIN`

for any relaying party.

the redirect_uri will only find use after the flow has completed

**Reply by vinckr** [reply]
Ok I see, and why is that not working?
What do you mean by `any relaying party` ?
Each login request should go to a different login uri? 
Or should some login request go to one uri and others to another? 

Could you give me some more information on the use case 🙏 ?

**Reply by wibed** [reply]
by relaying party I meant a hydra client.
atm `URLS_LOGIN` determines the login_uri for any relaying party on the hydra server.
 
what I am looking for is a way on how I could customise the login_uri for a relaying party(hydra client), without creating a new instance of hydra

**Reply by vinckr** [reply]
Hm I have to look a bit more, but I think this is not a feature at the moment.

But actually a good solution would be to spin up multiple instances of OryHydra with different `login_uri`! 
You can just copy the configuration and just change the `login_uri`. 
Ory Hydra is built with this in mind, so it is not some hack to deploy multiple instances. 

Let me know if that would work for you, maybe you can try it with two instances and then see if you can scale to your requirements.

**Reply by vinckr** [reply]
Hey @wibed 
I just got some clarification by @aeneasr 
> The proper way to handle that is to write a small application which decides the redirect URL based on the login request. 
> That app can then redirect to whatever UI they want to.

[![](https://mermaid.ink/img/eyJjb2RlIjoiZ3JhcGggVERcbkFbSHlkcmFdIC0tPkJ7bG9naW4gKnByb3h5Kn1cbiAgICBCIC0tPiB8cmVkaXJlY3R8Q1tMb2dpbiBVSSAxXVxuICAgIEIgLS0-IHxyZWRpcmVjdHxEW0xvZ2luIFVJIDJdXG4gICAgQiAtLT4gfHJlZGlyZWN0fEVbTG9naW4gVUkgbl1cbiAgICAgICAgICAgICIsIm1lcm1haWQiOnsidGhlbWUiOiJkZWZhdWx0In0sInVwZGF0ZUVkaXRvciI6ZmFsc2V9)](https://mermaid-js.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoiZ3JhcGggVERcbkFbSHlkcmFdIC0tPkJ7bG9naW4gKnByb3h5Kn1cbiAgICBCIC0tPiB8cmVkaXJlY3R8Q1tMb2dpbiBVSSAxXVxuICAgIEIgLS0-IHxyZWRpcmVjdHxEW0xvZ2luIFVJIDJdXG4gICAgQiAtLT4gfHJlZGlyZWN0fEVbTG9naW4gVUkgbl1cbiAgICAgICAgICAgICIsIm1lcm1haWQiOnsidGhlbWUiOiJkZWZhdWx0In0sInVwZGF0ZUVkaXRvciI6ZmFsc2V9)

**Comment by vinckr** [comment, marked as answer]
reposting here, so it does not get lost in the thread:

> The proper way to handle that is to write a small application which decides the redirect URL based on the login request.
>  That app can then redirect to whatever UI they want to.


[![](https://mermaid.ink/img/eyJjb2RlIjoiZ3JhcGggVERcbkFbSHlkcmFdIC0tPkJ7bG9naW4gKnByb3h5Kn1cbiAgICBCIC0tPiB8cmVkaXJlY3R8Q1tMb2dpbiBVSSAxXVxuICAgIEIgLS0-IHxyZWRpcmVjdHxEW0xvZ2luIFVJIDJdXG4gICAgQiAtLT4gfHJlZGlyZWN0fEVbTG9naW4gVUkgbl1cbiAgICAgICAgICAgICIsIm1lcm1haWQiOnsidGhlbWUiOiJkZWZhdWx0In0sInVwZGF0ZUVkaXRvciI6ZmFsc2V9)](https://mermaid-js.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoiZ3JhcGggVERcbkFbSHlkcmFdIC0tPkJ7bG9naW4gKnByb3h5Kn1cbiAgICBCIC0tPiB8cmVkaXJlY3R8Q1tMb2dpbiBVSSAxXVxuICAgIEIgLS0-IHxyZWRpcmVjdHxEW0xvZ2luIFVJIDJdXG4gICAgQiAtLT4gfHJlZGlyZWN0fEVbTG9naW4gVUkgbl1cbiAgICAgICAgICAgICIsIm1lcm1haWQiOnsidGhlbWUiOiJkZWZhdWx0In0sInVwZGF0ZUVkaXRvciI6ZmFsc2V9)

**Reply by wibed** [reply]
thank you
